### PR TITLE
docs: rm type hints in ignite.contrib.metrics (6)

### LINKED ISSUE #1665: Use autodoc_typehints = "description" in conf.py
## 📚 Documentation

EDITED HERE: As a help wanted issue, please, see below comment https://github.com/pytorch/ignite/issues/1665#issuecomment-783240506 on what remains to do here. Thanks 

If there are many arguments in class/function/method, it's a little hard to read/scan all arguments (due to typehints).
Although there is a `Parameters` section, there might be missing some arguments (rarely though).
And we are duplicating type hints work in both signatures and docstring. Writing once can make less work on authors and less type hints mismatch can occur.

If we can remove type hints, it can be a little easier to read, and showing type hints only in `Parameters` section.

For example:
Before:
Screen Shot 2021-02-21 at 22 52 16
After:
Screen Shot 2021-02-21 at 22 51 40

Changing to this docs style needs a little work. It needs to 
- add
```py
autoclass_content = "both"
autodoc_typehints = "description"  # new in sphinx 3.1
```
those lines in `conf.py`
- removing type hints in docstring
- leaving type hints only in signatures.

Let me know what you think.
cc: @vfdev-5 @sdesrozis @trsvchn

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1665 
fixes #1665 
https://deploy-preview-1688--pytorch-ignite-preview.netlify.app/contrib/metrics.html

Description: 
- remove type hints in ignite.contrib.metrics 
- add missing doc string

Check list:

- [ ] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-sphinx==3.2.1
+sphinx==3.4
```
Comment (ydcjeff) [inline docs/requirements.txt:1]:
```suggestion
```

**Review by vfdev-5 — APPROVED** [review]
LGTM! Thanks a lot @ydcjeff !

File: ignite/contrib/metrics/regression/manhattan_distance.py [hunk 2]
```diff
@@ -21,6 +21,16 @@ class ManhattanDistance(_BaseRegression):
 
     __ https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.DistanceMetric.html
 
+    Args:
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/manhattan_distance.py:24]: Maybe we should remove the `__init__` method ?

File: ignite/contrib/metrics/regression/median_absolute_error.py [hunk 3]
```diff
@@ -31,6 +31,15 @@ class MedianAbsoluteError(_BaseRegressionEpoch):
 
     __ https://arxiv.org/abs/1809.03006
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/median_absolute_error.py:33]: It misses `device` argument in `__init__`

File: ignite/contrib/metrics/regression/median_absolute_percentage_error.py [hunk 4]
```diff
@@ -31,6 +31,15 @@ class MedianAbsolutePercentageError(_BaseRegressionEpoch):
 
     __ https://arxiv.org/abs/1809.03006
 
+    Args:
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/median_absolute_percentage_error.py:34]: It misses `device` argument in `__init__`

File: ignite/contrib/metrics/regression/median_relative_absolute_error.py [hunk 5]
```diff
@@ -31,6 +31,15 @@ class MedianRelativeAbsoluteError(_BaseRegressionEpoch):
 
     __ https://arxiv.org/abs/1809.03006
 
+    Args:
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/median_relative_absolute_error.py:34]: It misses `device` argument in `__init__`

File: ignite/contrib/metrics/regression/median_absolute_percentage_error.py (hunk 4 above)
Comment (vfdev-5) [reply ignite/contrib/metrics/regression/median_absolute_percentage_error.py:34]: Actually, they all missing device here as contrib metrics were not yet "finished"

File: ignite/contrib/metrics/regression/r2_score.py [hunk 6]
```diff
@@ -20,6 +20,16 @@ class R2Score(_BaseRegression):
     - ``update`` must receive output of the form ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
     - `y` and `y_pred` must be of same shape `(N, )` or `(N, 1)` and of type `float32`.
 
+    Args:
```
Comment (sdesrozis) [inline ignite/contrib/metrics/regression/r2_score.py:23]: `__init__` useless ?

**Comment by sdesrozis** [comment]
@ydcjeff I left comments

File: ignite/contrib/metrics/regression/manhattan_distance.py (hunk 2 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/manhattan_distance.py:24]: we still need to do `super().__init__`, right?

File: ignite/contrib/metrics/regression/r2_score.py (hunk 6 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/r2_score.py:23]: we still need to do super().__init__, right?

File: ignite/contrib/metrics/regression/median_absolute_percentage_error.py (hunk 4 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/median_absolute_percentage_error.py:34]: some metrics have `device` argument, some didn't.
Should `device` be removed or left as it is ?

File: ignite/contrib/metrics/regression/manhattan_distance.py (hunk 2 above)
Comment (sdesrozis) [reply ignite/contrib/metrics/regression/manhattan_distance.py:24]: `_BaseRegression` does not implement `__init__` and inherits from `Metric`. If no `__init__` in this class, `__init__` from `Metric` is called. Here, this `__init__` is a useless proxy to the `__init__` from `Metric` class. What do you think ?

File: ignite/contrib/metrics/regression/r2_score.py (hunk 6 above)
Comment (sdesrozis) [reply ignite/contrib/metrics/regression/r2_score.py:23]: Same than above.

File: ignite/contrib/metrics/regression/median_absolute_percentage_error.py (hunk 4 above)
Comment (sdesrozis) [reply ignite/contrib/metrics/regression/median_absolute_percentage_error.py:34]: IMO `device` should be removed. The docstring must be related to the code even it is not yet perfect.

File: ignite/contrib/metrics/regression/manhattan_distance.py (hunk 2 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/manhattan_distance.py:24]: Thanks for pointing out. Changed

File: ignite/contrib/metrics/regression/median_absolute_percentage_error.py (hunk 4 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/median_absolute_percentage_error.py:34]: Fortunately, docstring and code matched perfectly. 😄

File: ignite/contrib/metrics/regression/median_absolute_error.py (hunk 3 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/median_absolute_error.py:33]: Removed `device`

File: ignite/contrib/metrics/regression/median_relative_absolute_error.py (hunk 5 above)
Comment (ydcjeff) [reply ignite/contrib/metrics/regression/median_relative_absolute_error.py:34]: removed `device`

File: ignite/contrib/metrics/regression/manhattan_distance.py (hunk 2 above)
Comment (sdesrozis) [reply ignite/contrib/metrics/regression/manhattan_distance.py:24]: What about adding a comment on that point in the docstring ? I mean that the arguments relie on `Metric` `__init__`

**Review by sdesrozis — APPROVED** [review]
@ydcjeff Thank you ! Looks good to me.
